src/mqtt.h

```diff
@@ -19,19 +19,32 @@
 #define MQTT_CMD_DISCONNECT 14
 #define MQTT_CMD_AUTH 15
 
+#define MQTT_PROPS_USER_PROPERTY_ID 0x26
+
 enum { MQTT_OK, MQTT_INCOMPLETE, MQTT_MALFORMED };
 
+struct mg_mqtt_property {
+  uint8_t id;     // Enumerated at MQTT5 Reference
+  uint32_t iv;  // integer value for 8-, 16-, 32-bit integers types
+  struct mg_str key;    // non-NULL only for user property type
+  struct mg_str val;    // non-NULL only for UTF-8 types and user properties
+};
+
 struct mg_mqtt_opts {
   struct mg_str user;          // Username, can be empty
   struct mg_str pass;          // Password, can be empty
   struct mg_str client_id;     // Client ID
-  struct mg_str will_topic;    // Will topic
-  struct mg_str will_message;  // Will message
-  uint8_t will_qos;            // Will message quality of service
+  struct mg_str topic;        // topic
+  struct mg_str message;      // message
+  uint8_t qos;                // message quality of service
   uint8_t version;             // Can be 4 (3.1.1), or 5. If 0, assume 4.
   uint16_t keepalive;          // Keep-alive timer in seconds
-  bool will_retain;            // Retain last will
+  bool retain;                // Retain last will
   bool clean;                  // Use clean session, 0 or 1
+  struct mg_mqtt_property *properties;  // array of properties (valid only for version 5)
```

Comment:
Rename to `struct mg_mqtt_prop *props;`

```c
struct mg_mqtt_prop *props; // MQTT5 props array, ended with ID 0 property
```

Reply:
From my understanding, you are suggesting that each array has an element with the id of zero. This means there is no reason for the field nr_properties to exist anymore. However, I think this might be a bit risky for the user, since, if they forget to add the property with ID 0, the program will most likely crash. On the other hand, setting the number of properties would seem more natural to the user and is also safer in my opinion. Let me know what you think about this.

Reply:
Yeah, you're right, using an explicit count is better than a sentinel. Let's go that way.